packages/@lwc/engine-core/src/framework/modules/props.ts

```diff
@@ -43,6 +43,20 @@ export function patchProps(
             isFirstPatch ||
             cur !== (isLiveBindingProp(sel, key) ? getProperty(elm!, key) : oldProps[key])
         ) {
+            // Additional verification if properties are supported by the element
+            // Validation relies on html properties and public properties being defined on the element,
+            // SSR has its own custom validation.
+            if (process.env.IS_BROWSER && process.env.NODE_ENV !== 'production') {
+                if (!(key in elm!)) {
+                    assert.fail(
+                        `Unknown public property "${key}" of element <${
+                            elm!.tagName
+                        }>. This is likely a typo on the corresponding attribute "${htmlPropertyToAttribute(
+                            key
+                        )}".`
+                    );
+                }
+            }
```

Comment:
Why was this moved and put behind a `process.env.IS_BROWSER` check?

Reply:
This validation relies on `key in elm` check to work, where key is handled by descriptors for the element's global html properties and public(`@api`) properties. These properties are only available on the element in a browser environment and not in SSR.